internal/host/hostcommands/domain_name_resolution_cmd.go

```diff
@@ -0,0 +1,97 @@
+package hostcommands
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/constants"
+	"github.com/openshift/assisted-service/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type domainNameResolutionCmd struct {
+	baseCmd
+	domainNameResolutionImage string
+	db                        *gorm.DB
+}
+
+func NewDomainNameResolutionCmd(log logrus.FieldLogger, domainNameResolutionImage string, db *gorm.DB) *domainNameResolutionCmd {
+	return &domainNameResolutionCmd{
+		baseCmd:                   baseCmd{log: log},
+		domainNameResolutionImage: domainNameResolutionImage,
+		db:                        db,
+	}
+}
+
+func (f *domainNameResolutionCmd) prepareParam(host *models.Host, cluster *common.Cluster) (string, error) {
+	clusterName := cluster.Cluster.Name
+	if len(clusterName) == 0 {
+		err := errors.Errorf("Cluster name is empty for cluster %s", host.ClusterID)
+		f.log.WithError(err).Warn("Cluster name is empty")
+		return "", err
+	}
+
+	baseDNSDomain := cluster.Cluster.BaseDNSDomain
+	if len(baseDNSDomain) == 0 {
+		err := errors.Errorf("Cluster base domain is empty for cluster %s", host.ClusterID)
+		f.log.WithError(err).Warn("Cluster base domain is empty")
+		return "", err
+	}
+	apiDomainName := fmt.Sprintf("api.%s.%s", clusterName, baseDNSDomain)
```

Comment:
we need external api resolution only in None platform with olm in all other cases it will not block the installation.
We still want to block installation cause of it?

Reply:
got an answer in slack thanks

Reply:
got an answer in slack, thanks